node/silkworm/db/access_layer.cpp

```diff
@@ -297,6 +307,15 @@ void parse_senders(mdbx::txn& txn, const Bytes& key, std::vector<Transaction>& o
     }
 }
 
+void write_senders(mdbx::txn& txn, const Bytes& key, const std::vector<Transaction>& transactions) {
+    std::vector<evmc::address> buffer;
+    for (const auto& transaction : transactions) {
+        SILKWORM_ASSERT(transaction.from);
+        buffer.push_back(*transaction.from);
+    }
+    write_senders_raw(txn, key, buffer);
+}
```

Comment:
Don't use a vector here. Use `Bytes` which is more meaningful for downstream call `write_senders_raw`. Vector might be reallocated during loop.
Also use an early exit check.

```suggestion
void write_senders(mdbx::txn& txn, const Bytes& key, const std::vector<Transaction>& transactions) {
    if(transactions.empty()) {
        return;
    }
    Bytes data(transactions.size() * kAddressLength, '\0');
    size_t data_offset{0};
    for(const auto& transaction:transactions) {
        SILKWORM_ASSERT(transaction.from.has_value());
        std::memcpy(&data[data_offset], transaction.from.value().bytes, kAddressLength);
        data_offset += kAddressLength;
    }
    write_senders_raw(txn, key, data);
}```

Reply:
Done